client/src/Actions/DepositActions.js

```diff
@@ -2,23 +2,38 @@ import axios from 'axios';
 import URL from './index';
 import * as types from './actionTypes';
 
-export const MakeDeposit = (deposit_details, token) => dispatch => {
-  let requestObject = {
-    amount: deposit_details.amount,
-    currency: deposit_details.currency,
-    deposit_method: deposit_details.deposit_method,
-    deposit_method_type: deposit_details.deposit_method_type,
-  }
+import { tokenIsValid } from './../Helpers/tokenIsValid'
 
+export const MakeDeposit = (deposit_details, token) => dispatch => {
   dispatch({
     type: types.DEPOSITING_START,
   })
+
+  const { amount, currency, deposit_method, deposit_method_type} = deposit_details;
+  if (!amount || !currency || deposit_method || deposit_method_type) {
+    return dispatch({ 
+      type: types.DEPOSITING_FAILURE, 
+      payload: {
+        error: "Specify the amount, currency, deposit_method, and deposit_method_type"
+      } 
+    });
+  }
+
+  if (!tokenIsValid(token)) {
+    return dispatch({ 
+      type: types.DEPOSITING_FAILURE, 
```

Comment:
A bit of repetition here. Is there a better way?

Reply:
I tried coming up with ways to avoid having to call the helper function repeatedly but couldn't come up with something. One thing I might be able to do is to create a separate handler function that takes the given actionType (i.e. `GETTING_USER`) as a parameter that then returns the dispatch with the corresponding type.

Reply:
Good idea 👍